packages/web/src/components/gcds-date-input/gcds-date-input.tsx

```diff
@@ -0,0 +1,542 @@
+      } else {
+        let splitValue = this.value.split('-');
+        this.yearValue = splitValue[0];
+        this.monthValue = splitValue[1];
+        this.dayValue = splitValue[2];
+      }
+    }
+  }
+
+  /**
+   * Format day input value to add 0 to single digit values
+   */
+  private formatDay(e) {
+    if (!isNaN(e.target.value) && e.target.value.length === 1) {
+      this.dayValue = '0' + e.target.value;
+    }
+  }
+
+  private validateRequiredProps() {
+    this.validateName();
+    this.validateLegend();
+    this.validateFormat();
+
+    if (
+      this.errors.includes('name') ||
+      this.errors.includes('legend') ||
+      this.errors.includes('format')
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  async componentWillLoad() {
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.validateValidator();
+
+    // Assign required validator if needed
+    requiredValidator(this.el, 'date-input');
+
+    if (this.validator) {
+      this._validator = getValidator(this.validator);
+    }
+
+    let valid = this.validateRequiredProps();
+
+    if (!valid) {
+      logError('gcds-date-input', this.errors);
+    }
+
+    this.validateValue();
+    if (this.value && isValidDate(this.value)) {
+      this.splitFormValue();
+      this.setValue();
+
+      this.initialValue = this.value;
+    }
+  }
+
+  componentWillUpdate() {
+    if (this.validator) {
+      this._validator = getValidator(this.validator);
+    }
+  }
+
+  render() {
+    const {
+      legend,
+      name,
+      format,
+      required,
+      hint,
+      errorMessage,
+      disabled,
+      lang,
+      hasError,
+    } = this;
+
+    let requiredAttr = {};
+
+    if (required) {
+      requiredAttr['aria-required'] = 'true';
+    }
+
+    // Array of months 01 - 12
+    const options = Array.from({ length: 12 }, (_, i) =>
+      i + 1 < 10 ? `0${i + 1}` : `${i + 1}`,
+    );
+
+    const month = (
+      <gcds-select
+        label={i18n[lang].month}
+        selectId="month"
+        name="month"
+        defaultValue={i18n[lang].selectmonth}
+        disabled={disabled}
+        onInput={e => this.handleInput(e, 'month')}
+        onChange={e => this.handleInput(e, 'month')}
+        value={this.monthValue}
+        class={`gcds-date-input__month ${hasError['month'] ? 'gcds-date-input--error' : ''}`}
+        {...requiredAttr}
+        aria-invalid={hasError['month'].toString()}
+        aria-description={hasError['month'] && errorMessage}
+      >
+        {options.map(option => (
+          <option key={option} value={option}>
+            {i18n[lang]['months'][option]}
+          </option>
+        ))}
+      </gcds-select>
+    );
+
+    const year = (
+      <gcds-input
+        name="year"
+        label={i18n[lang].year}
+        inputId="year"
+        type="number"
+        size={4}
+        disabled={disabled}
+        value={this.yearValue}
+        autocomplete="year"
```

Comment:
I chose to remove the autocomplete attributes since there is no way to assign them a value that will work across multiple use cases of the date input.